packages/flutter_test/lib/src/binding.dart

```diff
@@ -634,10 +634,14 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
     return testCompleter.future;
   }
 
+  void _runAppSync(Widget app) {
+    runApp(app);
+  }
+
   Future<void> _runTestBody(Future<void> testBody(), VoidCallback invariantTester) async {
     assert(inTest);
 
-    runApp(Container(key: UniqueKey(), child: _preTestMessage)); // Reset the tree to a known state.
+    _runAppSync(Container(key: UniqueKey(), child: _preTestMessage)); // Reset the tree to a known state.
```

Comment:
Why this change? It seems identical?

Reply:
oh, i see, you override it later

Reply:
This is really urgly, but i can't think of other way around it unless we add an api called scheduleAttachRootWidget, and override it in testbinding to make it synchronized